src/client/Microsoft.Identity.Client/AppConfig/PopAuthenticationConfiguration.cs

```diff
@@ -3,46 +3,99 @@
+    /// To use a hardware key or an external key, implement <see cref="PopCryptoProvider"/>.
+    /// </remarks>
 #if !SUPPORTS_CONFIDENTIAL_CLIENT
     [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
 #endif
     public class PoPAuthenticationConfiguration
     {
         /// <summary>
-        /// Constructs the configuration properties used to construct a proof of possession request
+        /// Creates a configuration using the default key management. Use the properties <see cref="HttpMethod"/>, <see cref="HttpHost"/> etc. 
+        /// to control which elements of the request should be included in the POP token.
         /// </summary>
-        /// <param name="requestUri">An Uri to the protected resource which requires a PoP token. The PoP token will be cryptographically bound to the request uri</param>
-        public PoPAuthenticationConfiguration(Uri requestUri)
+        /// <remarks>
+        /// See https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3 for details about signed HTTP requests.
+        /// </remarks>
+        public PoPAuthenticationConfiguration()
         {
             ConfidentialClientApplication.GuardMobileFrameworks();
-            RequestUri = requestUri ?? throw new ArgumentNullException(nameof(requestUri));
         }
 
         /// <summary>
-        /// An Uri to the protected resource which requires a PoP token. The PoP token will be cryptographically bound to the request uri (to the host and query parts of it)
+        /// Creates a configuration using the default key management, and which binds all the details of the HttpRequestMessage.
         /// </summary>
-        public Uri RequestUri { get; }
+        /// <remarks>
+        /// Currently only the HtppMethod (m), UrlHost (u) and UrlPath (p) are used to create the signed HTTP request - see https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3
+        /// </remarks>
+        public PoPAuthenticationConfiguration(HttpRequestMessage httpRequestMessage)
+        {
+            if (httpRequestMessage == null)
+                throw new ArgumentNullException(nameof(httpRequestMessage));
+
+            HttpMethod = httpRequestMessage.Method;
+            HttpHost = httpRequestMessage.RequestUri.Authority; // this includes the optional port
+            HttpPath = httpRequestMessage.RequestUri.AbsolutePath; 
+        }
+
+      
+
+        /// <summary>
+        /// Creates a configuration using the default key management, and which binds only the Uri part of the HTTP request.
+        /// </summary>
+        /// <remarks>
+        /// The UrlHost (u) and UrlPath (p) are used to create the signed HTTP request - see https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3
+        /// </remarks>
+        public PoPAuthenticationConfiguration(Uri requestUri)
```

Comment:
I think we should remove this API. The spec states that a POP request is valid even with NO Http Signed request claims. We just don't have E2E infrastructure to test it. I'll leave it for now and confirm with middleware team.

Reply:
deprecate it?

Reply:
API is "experimental", so we can delete it.

Reply:
How about deprecate for 1 release, then delete? API is experimental, we can delete, but I'm mindful of not breaking MIW.

Reply:
M.I.W can react quickly. right this is experimental..
I agree with the approach

Reply:
Will leave for now, let's see how this evolves.